main.tf

```diff
@@ -242,6 +251,15 @@ resource "aws_autoscaling_group" "this_ignore_desired_capacity_changes" {
     local.tags_asg_format,
   )
 
+  instance_refresh {
```

Comment:
it's better to use `dynamic` block here to not alter existing behaviour

maybe we can use `var.asg_instance_refresh_strategy` as the switch (and it should have "null" as default value)
```hcl
dynamic "instance_refresh" {
    for_each = var.asg_instance_refresh_strategy != null ? [1] : []
    content {
        strategy = var.asg_instance_refresh_strategy
        //continued...
        dynamic "preferences" {
              //continued for conditional override
        }
    }
}
```

Reply:
Thank for the suggestion, I will take a look at Dynamic block and adjust the changes